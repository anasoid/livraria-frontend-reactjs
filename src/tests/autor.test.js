import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Autor from '../components/autor';
import AutorContainer from '../components/autor/autor-conteiner';


describe('test Autor component', () => {
  it('renders without crashing', () => {
    shallow(<Autor/>);
    const tree = renderer.create(<Autor/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render autor AutorContainer', () => {
    const wrapper = shallow(<Autor/>);
    expect(wrapper.find(AutorContainer)).toExist();
  });
});