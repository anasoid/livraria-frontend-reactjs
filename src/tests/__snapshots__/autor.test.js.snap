// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Autor component renders without crashing 1`] = `
<div>
  <h1>
    Novo Autor
  </h1>
  <form>
    <label
      htmlFor="autorNome"
    >
      Nome: 
    </label>
    <input
      id="autorNome"
      name="nome"
    />
    <br />
    <label
      htmlFor="autorEmail"
    >
      Email: 
    </label>
    <input
      id="autorEmail"
      name="email"
    />
    <br />
    <button
      id="btnGravarAutor"
      type="submit"
    >
      Gravar Autor
    </button>
  </form>
  <br />
  <div>
    <table>
      <caption>
        Autores
      </caption>
      <thead>
        <tr>
          <th>
            Nome
          </th>
          <th>
            Email
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
